### ISSUE TITLE
t.LOCK.UPDATE not stopping two people findOne

### ISSUE THREAD

**Issue by PCFNode** [sequelize/sequelize#8196]
## What you are doing?

I have a table called User (for the sake of this example). When a person calls a function, it checks the users `status` and then, if the `status` is `Active`, goes through multiple database creates and updates, ending with updating the users `status` to `NotActive`

```
let t = await models.sequelize.transaction(); 
let getTheUser = await models.User.findOne({ where: { id: 21 }, lock: t.LOCK.UPDATE, transaction: t } );
if (getTheUser.status != "Active") {
  throw Error("This User is not active");
}
// onto code which does a bunch of stuff, then changes the User status to NotActive
```

## What do you expect to happen?
This is implementing a classic banking situation. I only want the code to run if the user is currently in `Active` status.  What we expect to happen is to lock the transaction and prevent the second user from continuing. 

## What is actually happening?
The transaction (and all the creates and updates after the find) is massive, so can take 30 seconds. When the call is done twice in that time, the second user also see's it as `Active` and is able to do all the subsequent database work. 
Until the transaction is committed, a second user can get past the findOne, still thinking the User is in the `Active` status.

## What I have also tried...
I am not the most clued up on databases. Above I used LOCK on the specific find query, assuming it would lock the row/table so the second user was unable to query until the transaction was completed. However, I have also tried changing the isolationLevel (in the sequelize config, as well as the specific transaction creation) and this did not work either

__Dialect:__ mssql
__Database version:__ Not sure, its an Azure SQL database
__Sequelize version:__ 3.29

**Comment by Alexsey** [comment]
In MySQL SELECT statement with WHERE with direct condition (like id = 21, opposite to id > 21) would lock only records that his the SELECT criteria no matter what isolation level you are using. BTW don't forget to ROLLBACK before `throw` - pending transactions in database is really not a good thing

**Comment by PCFNode** [comment]
It does lock the row, so when your debugging the application on two separate machines, the first one is allowed to read, then the second blocks until the first transaction is complete. However, as soon as the second one unblocks, its giving a result as if the first transaction never took place. What could I do to demonstrate this to you? As I have had three people look into this, and its replicable.

Also, the code sample above is just a snippet. After the `let t = ` I have a try/catch where the rollback is in the catch.

**Comment by Alexsey** [comment]
In this case first transaction should perform SELECT ... FOR UPDATE and have any isolation level while second transaction should have isolation level of READ COMMITED or higher. If it is the same code than both transaction should perform SELECT ... FOR UPDATE and have an isolation level of READ COMMITED or higher (if it is suitable for your other code). And looks like it is something you have done. I was having the common situation and it works for me... well, I could recommend you investigate the SQL logs that Sequelize generates to you and check the SELECT statements and isolation levels of your transaction

BTW try to set `AUTOCOMMIT` to `false` (`0`). There is not a lot of info on it in Sequelize docs. Just make sure your transactions are running with `AUTOCOMMIT = 0` (default value is `0`)

If this wouldn't help - I think you will have to write the minimal code that reproduces the problem and open another issue with better formalization - that would probably get a better chance to get more attention

If you will solve your problem - please write about it here
